packages/expo-image-manipulator/src/ExpoImageManipulator.web.ts

```diff
@@ -1,192 +1,9 @@
-          weightsAlpha += weight;
-          //colors
-          if (data[xPosition + 3] < 255) {
-            weight = (weight * data[xPosition + 3]) / 250;
-          }
-          gx_r += weight * data[xPosition];
-          gx_g += weight * data[xPosition + 1];
-          gx_b += weight * data[xPosition + 2];
-          weights += weight;
-        }
-      }
-      data2[x2] = gx_r / weights;
-      data2[x2 + 1] = gx_g / weights;
-      data2[x2 + 2] = gx_b / weights;
-      data2[x2 + 3] = gx_a / weightsAlpha;
-    }
-  }
-  //clear and resize canvas
-  if (resizeCanvas) {
-    canvas.width = width;
-    canvas.height = height;
-  } else {
-    ctx.clearRect(0, 0, widthSource, heightSource);
-  }
-
-  //draw
-  ctx.putImageData(img2, 0, 0);
-}
-
-function sizeFromAngle(
-  width: number,
-  height: number,
-  angle: number
-): { width: number; height: number } {
-  const radians = (angle * Math.PI) / 180;
-  let c = Math.cos(radians);
-  let s = Math.sin(radians);
-  if (s < 0) {
-    s = -s;
-  }
-  if (c < 0) {
-    c = -c;
-  }
-  return { width: height * s + width * c, height: height * c + width * s };
-}
-
-function cropImage(
-  canvas: HTMLCanvasElement,
-  image: HTMLImageElement,
-  x: number = 0,
-  y: number = 0,
-  width: number = 0,
-  height: number = 0
-): void {
-  const context = getContext(canvas);
-  context.save();
-  context.drawImage(image, x, y, width, height, 0, 0, width, height);
-}
-
-function drawImage(
-  canvas: HTMLCanvasElement,
-  img: HTMLImageElement,
-  x: number = 0,
-  y: number = 0,
-  angle: number = 0,
-  xFlip: boolean = false,
-  yFlip: boolean = false,
-  width?: number,
-  height?: number
-): CanvasRenderingContext2D {
-  const context = getContext(canvas);
-  context.save();
-
-  if (width == null) {
-    width = img.naturalWidth as number;
-  }
-  if (height == null) {
-    height = img.naturalHeight as number;
-  }
-
-  // Set the origin to the center of the image
-  context.translate(x + canvas.width / 2, y + canvas.height / 2);
-
-  // Rotate the canvas around the origin
-  const radians = (angle * Math.PI) / 180;
-  context.rotate(radians);
-
-  // Flip/flop the canvas
-  const xScale = xFlip ? -1 : 1;
-  const yScale = yFlip ? -1 : 1;
-
-  context.scale(xScale, yScale);
-
-  // Draw the image
-  context.drawImage(
-    img,
-    -img.naturalWidth / 2,
-    -img.naturalHeight / 2,
-    img.naturalWidth,
-    img.naturalHeight
-  );
-
-  context.restore();
-  return context;
-}
-
-function getContext(canvas: HTMLCanvasElement): CanvasRenderingContext2D {
-  const ctx = canvas.getContext('2d');
-  if (!ctx) {
-    throw new CodedError('ERR_IMAGE_MANIPULATOR', 'Failed to create canvas context');
-  }
-  return ctx;
-}
+import { ImageResult, SaveOptions, Action } from './ImageManipulator.types';
+import crop from './actions/CropAction.web';
+import flip from './actions/FlipAction.web';
+import resize from './actions/ResizeAction.web';
+import rotate from './actions/RotateAction.web';
```

Comment:
What would say about creating single `actions/index.ts` re-exporting all the actions from the respective files? 🤔 
And here we would have:
```suggestion
import  { crop, flip, resize, rotate } from './actions';
```

Reply:
Thanks for the feedback 😀 

I think that's a little neater I've added a commit 0b7d8226829300a26c7cdb42c6bc88c4845f6c28 implementing this (slightly different to the suggestion)

There's a slight error in your snack, I think I'd do`const [time, setTime] = React.useState(-1);`.
The reason I didnt include it in the snack with many actions is I think you end up with all the components competing for processing time, I've added a mutex implementation that I think should make only one run at a time.
To be honest the timing isn't much use on the relatively small images since it happens so fast.